crates/miden-objects/src/note/recipient.rs

```diff
@@ -59,10 +59,11 @@ impl NoteRecipient {
 
     /// Returns the recipient encoded as [Felt]s.
     pub fn to_elements(&self) -> Vec<Felt> {
-        let mut result = Vec::with_capacity(12);
+        let mut result = Vec::with_capacity(13);
         result.extend(self.inputs.commitment());
         result.extend(self.script.root());
         result.extend(self.serial_num);
+        result.push(self.inputs.num_values().into());
```

Comment:
Just an optional nit: I would move the number of inputs to be the first in this vector to have `[inputs_len, inputs_commitment, ... ]`. It won't really affect anything, just a desire to have a related values stored alongside.